org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java

```diff
@@ -0,0 +1,605 @@
+	 * 		.have(serviceReferenceIsNotNull())
+	 * 		.filteredOn(serviceReferenceIsNotNull())
+	 * 		.first()// map to {@link
+	 * 				// ObjectAssert}
+	 * 		.has(serviceReferenceIsNotNull());// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<ServiceEvent> serviceReferenceIsNotNull() {
+		return not(serviceReferenceIsNull());
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link ServiceEvent}.
+	 * Checking if the serviceReference of the {@link ServiceEvent} <b>is
+	 * null</b>.
+	 *
+	 * <pre>
+	 * List<ServiceEvent> serviceEvent = null;
+	 *
+	 * static void example_serviceReferenceIsNull() {
+	 *
+	 * 	assertThat(serviceEvent)// created an {@link ListAssert}
+	 * 		.have(serviceReferenceIsNull())
+	 * 		.filteredOn(serviceReferenceIsNull())
+	 * 		.first()// map to {@link
+	 * 				// ObjectAssert}
+	 * 		.has(serviceReferenceIsNull());// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<ServiceEvent> serviceReferenceIsNull() {
+
+		return VerboseCondition.verboseCondition((sericeEvent) -> sericeEvent.getServiceReference() == null,
+			"serviceReference is <null>", (se) -> se.getServiceReference()
+				.toString());
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link ServiceEvent}.
+	 * Checking if a given type-mask <b>matches</b> the type.
+	 *
+	 * <pre>
+	 * List<ServiceEvent> serviceEvents = null;
+	 *
+	 * static void example_type(int typeMask) {
+	 *
+	 * 	assertThat(serviceEvents)// created an {@link ListAssert}
+	 * 		.have(type(type))
+	 * 		.filteredOn(type(typeMask))
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(type(typeMask));// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param typeMask - the typeMask that would be checked against the bundle
+	 *            type of the {@link ServiceEvent}
+	 * @return the Condition
+	 */
+	static Condition<ServiceEvent> type(final int eventTypeMask) {
+
+		return VerboseCondition.verboseCondition(
+			(ServiceEvent se) -> Bitmap.typeMatchesMask(se.getType(), eventTypeMask),
+			"type matches mask <" + ServiceEventType.BITMAP.maskToString(eventTypeMask) + ">", //
+			se -> " but was <" + ServiceEventType.BITMAP.toString(se.getType()) + ">");
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link ServiceEvent}.
+	 * Checking if type of {@link ServiceEvent} <b>matches</b>
+	 * <b>ServiceEvent.MODIFIED</b>.
+	 *
+	 * <pre>
+	 * List<ServiceEvent> frameworkEvents = null;
+	 *
+	 * static void example_typeModified() {
+	 *
+	 * 	assertThat(serviceEvents)// created an {@link ListAssert}
+	 * 		.have(typeModified())
+	 * 		.filteredOn(typeModified())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(typeModified());// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<ServiceEvent> typeModified() {
```

Comment:
Same here. Not a method for each event type. It makes the API big and thus make a large testing surface for our tests.

Reply:
reduced to just one `type(int mask)` .